server/executor/poller_executor_test.go

```diff
@@ -0,0 +1,380 @@
+package executor_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/kubeshop/tracetest/server/config"
+	"github.com/kubeshop/tracetest/server/executor"
+	"github.com/kubeshop/tracetest/server/id"
+	"github.com/kubeshop/tracetest/server/model"
+	"github.com/kubeshop/tracetest/server/testdb"
+	"github.com/kubeshop/tracetest/server/tracedb"
+	"github.com/kubeshop/tracetest/server/tracedb/connection"
+	"github.com/kubeshop/tracetest/server/tracing"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+var (
+	randomIDGenerator = id.NewRandGenerator()
+)
+
+func Test_PollerExecutor_ExecuteRequest_NoRootSpan_NoSpanCase(t *testing.T) {
+	t.Parallel()
+
+	// Scenario: Trace without spans
+	// Given the trigger execution returns 0 spans
+	// And tracetest does not send the root span
+	// When the server do the polling process
+	// Then it should stop at the second iteration
+	// And it should have no error on the process
+	// And a root span should be added to it
```

Comment:
I think this should never happen. I think we should timeout the polling process instead of stopping it when we don't get any trace data.